### PR TITLE
Fix bug where no bonus XP assignments would display additional XP message

### LINKED ISSUE #1950: Contest submission mentions bonus XP
When finalising a submission for a contest, the popup text is similar to that of missions, quests & paths. It talks about bonus XP, even though [it has been said](https://piazza.com/class/krvh5l6ipib380?cid=111) that there is none.

### PR DESCRIPTION
**Pull request by jayjay19630**
### Description

This pull request fixes a bug where no bonus XP assignments would display additional XP messages upon finalisation. This was done by adding a new attribute to AssessmentOverview named 'earlySubmissionXp' and conditionally rendering finalisation messages depending on this attribute. Tests and assessment mocks were correspondingly updated. This resolves #1950 

### Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

### Checklist

- [x] I have tested this code

### CODE REVIEW

**Comment by jayjay19630** [comment]
Because a new attribute is being added to AssessmentOverviews, a corresponding backend PR is needed. Will open backend PR soon.

**Comment by jayjay19630** [comment]
Corresponding backend PR: [source-academy/backend/pull/1076](https://github.com/source-academy/backend/pull/1076)

File: src/commons/assessment/Assessment.tsx [hunk 1]
```diff
@@ -349,6 +349,21 @@ const Assessment: React.FC<AssessmentProps> = props => {
     );
   }
 
+  // Define the warning text when finalising submissions
+  const hasBonusXp = (betchaAssessment?.earlySubmissionXp as number) > 0;
+  console.log(hasBonusXp);
```
Comment (RichDom2185) [inline src/commons/assessment/Assessment.tsx:354]: Let's remove this console.log statement

**Review by RichDom2185 — CHANGES_REQUESTED** [review]
Hi, just the one minor comment, thanks!

File: src/commons/assessment/Assessment.tsx (hunk 1 above)
Comment (jayjay19630) [reply src/commons/assessment/Assessment.tsx:354]: sorry, i can't believe i missed that. will fix right away

Comment (RichDom2185) [reply src/commons/assessment/Assessment.tsx:354]: No worries, thanks!

**Review by RichDom2185 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**